MoreLinq.Test/TakeLastTest.cs

```diff
@@ -57,5 +66,22 @@ public void TakeLastDisposesSequenceEnumerator()
                 seq.TakeLast(1).Consume();
             }
         }
+
+        [Test]
+        public void TakeLastOptimizedForCollections()
+        {
+            var sequence = new UnenumerableList<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+
+            sequence.TakeLast(3).AssertSequenceEqual(8, 9, 10);
```

Comment:
Here we are assuming that the final implementation will indeed skip enumerating. Are we confident all framework versions will ever do that, and consider it broken until they do? Or maybe we should just test that TakeLast did its thing, which is to check the Count property before enumerating?

Reply:
Not sure what you mean. The tests are now checking against all supported frameworks. Also, [`TakeLast` now internally relies on our `Slice`](https://github.com/morelinq/MoreLINQ/pull/304/commits/5eb30a2d70fbfaa294f2c23cb07085f9cd670441#diff-46c4b502587f8dc7793481ad0dbc74e8R55) (since 5eb30a2d70fbfaa294f2c23cb07085f9cd670441) instead of `Skip` so we are more sure about what's going to happen.

Reply:
Oh, right. I somehow missed this change, so scratch my comment.

LGTM

Reply:
@fsateler Cool and thanks for the one last look nonetheless. :+1: